src/index.ts

```diff
@@ -1,102 +1,102 @@
-     return index * 2 + 1;
-   }
+  private maxHeap(elementOneIndex: number, elementTwoIndex: number) {
+    return this.extractor(this.array[elementOneIndex]) > this.extractor(this.array[elementTwoIndex]);
+  }
 
-   private right(index:number) {
-     return index * 2 + 2;
-   }
+  private minHeap(elementOneIndex: number, elementTwoIndex: number) {
+    return this.extractor(this.array[elementOneIndex]) < this.extractor(this.array[elementTwoIndex]);
+  }
 
-   private maxHeap(elementOneIndex: number, elementTwoIndex:number) {
-     return this.extractor(this.array[elementOneIndex]) > this.extractor(this.array[elementTwoIndex]);
-   }
+  private swap(indexOne: number, indexTwo: number) {
+    const tmp = this.array[indexOne];
+    this.array[indexOne] = this.array[indexTwo];
+    this.array[indexTwo] = tmp;
+  }
 
-   private minHeap(elementOneIndex: number, elementTwoIndex:number) {
-     return this.extractor(this.array[elementOneIndex]) < this.extractor(this.array[elementTwoIndex]);
-   }
+  private heapify(index: number) {
+    let currentIndex = index;
+    let parentIndex = this.parent(currentIndex);
 
-   private swap(indexOne:number, indexTwo:number) {
-     const tmp = this.array[indexOne];
-     this.array[indexOne] = this.array[indexTwo];
-     this.array[indexTwo] = tmp;
-   }
+    while (currentIndex && this.comparator(currentIndex, parentIndex)) {
+      this.swap(currentIndex, parentIndex);
 
-   private heapify(index: number) {
-     let currentIndex = index;
-     let parentIndex = this.parent(currentIndex);
+      currentIndex = parentIndex;
+      parentIndex = this.parent(parentIndex);
+    }
+  }
 
-     while (currentIndex && this.comparator(currentIndex, parentIndex)) {
-       this.swap(currentIndex, parentIndex);
+  private reBalance(index: number) {
+    let top = index;
+    const left = this.left(top);
+    const right = this.right(top);
 
-       currentIndex = parentIndex;
-       parentIndex = this.parent(parentIndex);
-     }
-   }
+    if (left < this.array.length && this.comparator(left, top)) { top = left; }
+    if (right < this.array.length && this.comparator(right, top)) { top = right; }
 
-   private reBalance(index:number) {
-     let top = index;
-     const left = this.left(top);
-     const right = this.right(top);
+    if (top !== index) {
+      this.swap(top, index);
+      this.reBalance(top);
+    }
+  }
 
-     if (left < this.array.length && this.comparator(left, top)) { top = left; }
-     if (right < this.array.length && this.comparator(right, top)) { top = right; }
+  private insertMany(iterator?: IterableIterator<T>) {
+    const loopCondition = true;
+    while (loopCondition) {
+      const element = iterator?.next();
 
-     if (top !== index) {
-       this.swap(top, index);
-       this.reBalance(top);
-     }
-   }
+      if (!element || element?.done) { break; }
+
+      this.array.push(element?.value);
+      this.heapify(this.array.length - 1);
+    }
+  }
```

Comment:
Actually, this can be public, so users can insert by bulk if they want to. Change it to public and move it up to the public methods sections (right under pop). And please add tests